lib/teleterm/api/proto/v1/gateway.proto

```diff
@@ -35,6 +35,9 @@ message Gateway {
     string target_uri = 3;
     // target_user is the target user
     string target_user = 4;
+    // target_subresource_name points at a subresource of the remote resource, for example a
+    // database name on a database server.
+    string target_subresource_name = 9;
```

Comment:
Picking the name wasn't easy. `target_name` is already reserved for as the name for the database server you're connecting to. On top of that, the fields in the gateway code in `lib/teleterm` try to not have db-specific names, so I couldn't name it just `target_database_name`.

I think we wanted to avoid adding db-specific stuff there in case we want to use proxies for more than databases, but we'll see how it'll actually pan out in the future.

Reply:
> … it would be more readable to aggregate those field in oneOf or create a separate message holding those fields instead of handling them by generic naming 

I was thinking of the second option too. It's almost like designing a database table, in a way.

The current implementation is a shot in the dark to be honest, it should become more clear what kind of stuff we need to share and what not once we actually get to implementing some other kinds of proxies.

> Also can we keep proto filed number in ascending order ?

`message Gateway` already has fields 1–8 and I wanted to add 9th. Instead of putting it as the last field though, I put it next to a field that's more closely related to it. From your experience, do you generally append them to the end?